DisguiseUnityRenderStream/Runtime/DisguiseCameraCapture.cs

```diff
@@ -23,67 +29,107 @@ public void Start()
             m_cameraData = new CameraData();
 
             m_camera = GetComponent<Camera>();
-            StreamDescription stream = Array.Find(m_RenderStream.Streams, s => s.name == gameObject.name);
-            m_frameSender = new FrameSender(gameObject.name, m_camera, stream);
-            RenderPipelineManager.endFrameRendering += RenderPipelineManager_endFrameRendering;
+            var stream = Array.Find(m_RenderStream.Streams, s => s.name == gameObject.name);
+            m_frameSender = new FrameSender(gameObject.name, stream);
+
+            m_capture = gameObject.AddComponent<CameraCapture>();
+            m_capture.hideFlags |= HideFlags.HideAndDontSave;
+
+            m_capture.description = m_frameSender.description;
+            m_capture.onCapture += OnCapture;
+        }
+
+        void OnEnable()
+        {
+            m_capture.enabled = true;
         }
 
-        // Update is called once per frame
-        public void Update()
+        void OnDisable()
         {
-            // set tracking
-            if (m_RenderStream == null) return;
+            m_capture.enabled = false;
+        }
+
+        // TODO: inject somewhere early in the player loop?
```

Comment:
We could apply the Disguise frame info onto the camera as soon as we receive it. The user would then be able to read the Camera params as early as `Update()` in their scripts. But per convention users do it in `LateUpdate()` and later anyway, there's not much to gain. Removed the comment.

Reply:
Now I'm thinking that it should happen in LateUpdate (or even later) because we don't want users messing with the cameras before rendering out to disguise.